src/portal/lib/src/config/system/system-settings.component.ts

```diff
@@ -60,9 +95,115 @@ export class SystemSettingsComponent {
         return this.hasAdminRole && this.hasCAFile;
     }
 
-    constructor( @Inject(SERVICE_CONFIG) private configInfo: IServiceConfig) {
+    /**
+     *
+     * Save the changed values
+     *
+     * @memberOf ConfigurationComponent
+     */
+    public save(): void {
+        let changes = this.getChanges();
+        if (!isEmpty(changes)) {
+            this.onGoing = true;
+            this.configService.saveConfigurations(changes)
+                .then(response => {
+                    this.onGoing = false;
+                    // API should return the updated configurations here
+                    // Unfortunately API does not do that
+                    // To refresh the view, we can clone the original data copy
+                    // or force refresh by calling service.
+                    // HERE we choose force way
+                    this.retrieveConfig();
+                    if ('read_only' in changes) {
+                        this.readOnlyChange.emit(changes['read_only']);
+                    }
+
+                    this.reloadSystemConfig.emit();
+                    this.errorHandler.info('CONFIG.SAVE_SUCCESS');
+                })
+                .catch(error => {
+                    this.onGoing = false;
+                    this.errorHandler.error(error);
+                });
+        } else {
+            // Inprop situation, should not come here
+            console.error('Save abort because nothing changed');
+        }
+    }
+
+    retrieveConfig(): void {
+        this.onGoing = true;
+        from(toPromise(this.configService.getConfigurations()))
+            .subscribe((configurations: Configuration) => {
+                this.onGoing = false;
+                // Add two password fields
+                configurations.email_password = new StringValueItem(fakePass, true);
+                this.config = configurations;
+                // Keep the original copy of the data
+                this.originalConfig = clone(configurations);
+            }, error => {
+                this.onGoing = false;
+                this.errorHandler.error(error);
+            });
+    }
+
+    reset(changes: any): void {
+        if (!isEmpty(changes)) {
+            for (let prop in changes) {
+                if (this.originalConfig[prop]) {
+                    this.config[prop] = clone(this.originalConfig[prop]);
+                }
+            }
+        } else {
+            // force reset
```

Comment:
if changes are empty, why force retrieve config?

Reply:
I did not add this logic. It just moved from config.component.ts